snapshot/git/gitdb/local_data.go

```diff
@@ -0,0 +1,71 @@
+package gitdb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/scootdev/scoot/snapshot"
+)
+
+const localIDText = "local"
+const localIDFmt = "%s-%s-%s"
+
+// localValue holds a reference to a value that is in the local DB
+type localValue struct {
+	sha  string
+	kind valueKind
+}
+
+const (
+	snapshotIDText            = "snap"
+	snapshotWithHistoryIDText = "swh"
+)
+
+var kindToIDText = map[valueKind]string{
+	kindSnapshot:            snapshotIDText,
+	kindSnapshotWithHistory: snapshotWithHistoryIDText,
+}
+
+var kindIDTextToKind = map[string]valueKind{
+	snapshotIDText:            kindSnapshot,
+	snapshotWithHistoryIDText: kindSnapshotWithHistory,
+}
+
+func (v *localValue) ID() snapshot.ID {
+	return snapshot.ID(fmt.Sprintf(localIDFmt, localIDText, kindToIDText[v.kind], v.sha))
+}
+
+// parseID parses ID into a localValue
+func parseID(id snapshot.ID) (*localValue, error) {
+	if id == "" {
+		return nil, fmt.Errorf("empty snapshot ID")
+	}
+	if !strings.HasPrefix(string(id), localIDText+"-") {
+		return nil, fmt.Errorf("unrecognized snapshot ID scheme in %s", id)
+	}
+	parts := strings.Split(string(id), "-")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("did not scan 3 items from %s", id)
+	}
+	kindText, sha := parts[1], parts[2] // skip parts[0] because we know it's local
```

Comment:
I'd prefer to do that check as part of all the post-split checks but no big deal.